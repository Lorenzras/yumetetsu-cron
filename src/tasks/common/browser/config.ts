
export const browserURL = 'http://127.0.0.1:9222';
export const optionsTest = () => ({
  devTools: true,
  headless: false,
  slowMo: 300,
});
export const browserTimeOut = 30000;
