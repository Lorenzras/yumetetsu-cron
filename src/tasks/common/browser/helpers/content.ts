
import {logger} from '../../../../utils';
import {Page} from 'puppeteer';

export const getSelectOptions = async (page : Page, selector: string) => {
  logger.info(`Retrieving options of ${selector}`);
  await page.waitForSelector(selector);

  const allOptions = await page.$$eval(
    selector + ' > option', (options) => {
      return (options as unknown as HTMLOptionElement[]).map(
        (option) => ({value: option.value, text: option.textContent}),
      );
    },
  );

  return allOptions.filter((opt) => opt.text?.length);
};
