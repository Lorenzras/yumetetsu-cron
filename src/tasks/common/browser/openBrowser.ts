import puppeteer, {Browser} from 'puppeteer';
import {logger} from '../../../utils';
import {optionsTest, browserURL} from './config';

const isTest = process.env.NODE_ENV === 'test';

const options = isTest ? optionsTest() : undefined;

const getPage = async (browser: Browser) => {
  const pages = await browser.pages();
  return pages.length > 0 ? pages[0] : browser.newPage();
};

export const launchBrowser = () => {
  logger.info('Launching browser.');
  return puppeteer.launch(options);
};

export const openBrowserPage = async () => {
  logger.info('Opening page.');
  const browser = await launchBrowser();
  return getPage(browser);
};

/**
 * For testing
 *
 * @return {Promise<puppeteer.Page>} Page
 */
export const openMockBrowserPage = async () => {
  logger.info('Opening mock browser page.');
  const browser = await puppeteer.connect({
    browserURL: 'http://127.0.0.1:9222',
    // browserWSEndpoint: 'ws://127.0.0.1:9222/devtools/browser/87ef8a6a-0405-4c36-821f-309ad007f026',
    defaultViewport: null});
  return getPage(browser);
};
