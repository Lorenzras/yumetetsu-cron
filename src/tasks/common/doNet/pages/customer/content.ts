import {Page} from 'puppeteer';
import {getSelectOptions} from '../../../browser/helpers/content';
import selectors from './selectors';
import {logger} from '../../../../../utils';

export const getOptionsStore = ( page: Page) => {
  return getSelectOptions(page, selectors.ddStores);
};

export const getOptionsEmployee = (page: Page) => {
  return getSelectOptions(page, selectors.ddAgents);
};


/**
 * Retrieves count.
 * Requires customer screen after search button has been clicked.
 *
 * @export
 * @param {Page} page
 * @return {Promise<number>} s
 */
export async function getResultCount(page: Page): Promise<number> {
  const count = await page.$eval(
    selectors.resultCount +
    ' span:first-child', (e) => {
      return (e as unknown as HTMLSpanElement).innerText;
    }).catch(() => 0);
  logger.info(`Search result has ${count} records `);

  return +count;
}

