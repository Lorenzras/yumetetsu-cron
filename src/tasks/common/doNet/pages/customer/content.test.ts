import {openMockBrowserPage} from '../../../browser';
import {getOptionsEmployee, getOptionsStore} from './content';

describe('Content', ()=>{
  it('retrieved', async ()=>{
    const page = await openMockBrowserPage();
    const storeOptions = await getOptionsStore(page);
    const employeeOptions = await getOptionsEmployee(page);

    page.browser().disconnect();

    expect(storeOptions).toMatchSnapshot();
    expect(employeeOptions).toMatchSnapshot();
  }, 30000);
});
