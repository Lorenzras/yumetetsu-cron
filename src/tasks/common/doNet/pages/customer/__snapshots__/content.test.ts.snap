// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content retrieved 1`] = `
Array [
  Object {
    "text": "ハウスドゥ！豊田中央店",
    "value": "157",
  },
  Object {
    "text": "ハウスドゥ！豊川中央店",
    "value": "270",
  },
  Object {
    "text": "ハウスドゥ  豊橋向山",
    "value": "403",
  },
  Object {
    "text": "ハウスドゥ  千種大久手",
    "value": "416",
  },
  Object {
    "text": "ハウスドゥ  高浜中央",
    "value": "488",
  },
  Object {
    "text": "ハウスドゥ！豊田美里店　仮称",
    "value": "575",
  },
  Object {
    "text": "ハウスドゥ！豊田大林店",
    "value": "578",
  },
  Object {
    "text": "ハウスドゥ！豊川八幡店",
    "value": "808",
  },
  Object {
    "text": "ハウスドゥ  家・不動産買取専門店  豊橋向山",
    "value": "897",
  },
  Object {
    "text": "ハウスドゥ  大垣",
    "value": "942",
  },
  Object {
    "text": "ハウスドゥ！中川八熊店",
    "value": "991",
  },
  Object {
    "text": "ハウスドゥ！蒲郡店",
    "value": "1155",
  },
  Object {
    "text": "ハウスドゥ！日進店",
    "value": "1264",
  },
  Object {
    "text": "ハウスドゥ！豊橋Ａ店　仮称",
    "value": "1305",
  },
  Object {
    "text": "ハウスドゥ  豊橋藤沢",
    "value": "1343",
  },
]
`;

exports[`Content retrieved 2`] = `
Array [
  Object {
    "text": "選択してください",
    "value": "",
  },
  Object {
    "text": "未設定",
    "value": "null",
  },
  Object {
    "text": "神谷昌孝",
    "value": "1556",
  },
  Object {
    "text": "メール",
    "value": "1814",
  },
  Object {
    "text": "神谷　昌孝",
    "value": "5647",
  },
  Object {
    "text": "山豊工建",
    "value": "9381",
  },
  Object {
    "text": "國安　真帆",
    "value": "9513",
  },
  Object {
    "text": "石川　圭子",
    "value": "12782",
  },
  Object {
    "text": "小出　絵美",
    "value": "13004",
  },
  Object {
    "text": "水野　一徳",
    "value": "13686",
  },
  Object {
    "text": "吉村　真優",
    "value": "14384",
  },
  Object {
    "text": "影島　湧人",
    "value": "16017",
  },
  Object {
    "text": "水谷　共輔",
    "value": "16146",
  },
  Object {
    "text": "中邨 宏映",
    "value": "16681",
  },
  Object {
    "text": "長木 未樹",
    "value": "16695",
  },
  Object {
    "text": "物件登録専用③",
    "value": "16696",
  },
  Object {
    "text": "立川　チアゴ",
    "value": "16890",
  },
  Object {
    "text": "RPAテスト",
    "value": "16897",
  },
  Object {
    "text": "嘉代　大河",
    "value": "17443",
  },
  Object {
    "text": "西濱　春輝",
    "value": "18480",
  },
  Object {
    "text": "平賀　大翔",
    "value": "18481",
  },
  Object {
    "text": "齊藤	美有",
    "value": "20857",
  },
  Object {
    "text": "加地	由樹子",
    "value": "20858",
  },
  Object {
    "text": "本田　勇大",
    "value": "20923",
  },
  Object {
    "text": "品川　博文",
    "value": "20924",
  },
]
`;
